packages/dd-trace/src/encode/chunk.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const DEFAULT_MIN_SIZE = 2 * 1024 * 1024 // 2MB
+
+class Chunk {
+  constructor (minSize = DEFAULT_MIN_SIZE) {
+    this.buffer = Buffer.allocUnsafe(minSize)
+    this.length = 0
+    this._minSize = minSize
+  }
+
+  push (...bytes) {
+    this._reserve(bytes.length)
+
+    for (const byte of bytes) {
+      this.buffer[this.length++] = byte
+    }
+  }
+
+  write (value) {
```

Comment:
The idea is that it's basically the same as `Buffer.write`. In that sense, it could support more types down the line (there was even an implementation that didn't reach the PR that actually did). The purpose of this class is that it can mimic the interface of both `Array` and `Buffer` to reduce code changes in the encoder.

Reply:
Got it. Please indicate this somewhere.